### PR TITLE
treewide: cleanups preparing release

### PR DESCRIPTION
**Pull request by PolynomialDivision**

### CODE REVIEW

**Comment by john-tho** [comment]
A few extras still around:
```
find target/linux -iname '*-5.10'
target/linux/ipq806x/files-5.10
target/linux/bcm4908/files-5.10
target/linux/realtek/dts-5.10
target/linux/realtek/files-5.10
target/linux/generic/pending-5.10
target/linux/generic/hack-5.10
target/linux/generic/backport-5.10
```

The make `DEPENDS`ed and config vars `LINUX_5_10` & `LINUX_5_15`
`git grep -E 'LINUX_5_1(0|5)'`

https://github.com/openwrt/openwrt/pull/12531/commits/b54c6617f881239918f1ef201644921c832fc7fc
>Remove the old 5.10 patches and configs except ath25, lantiq and omap

The commit message is not correct? Those paths got removed as well.

**Comment by john-tho** [comment]
There are a few [in-tree drivers that use `KERNEL_VERSION`](https://github.com/search?q=repo%3Aopenwrt%2Fopenwrt+LINUX_VERSION_CODE+path%3A%2F%5Etarget%5C%2Flinux%5C%2F%2F&type=code) (and an awful lot of patches), but it might be more work than it is worth to adjust them. I guess we could come up with a https://coccinelle.lip6.fr/ semantic patch for the C files. Worth waiting to hear some more opinions before starting on that.

**Comment by guidosarducci** [comment]
> > Also, it might be premature to drop the LINUX_5_15 references, and instead review them as part of kernel 6.1 integration work.
> 
> Could you please explain this in more detail? I am not sure what the advantages are of keeping the references?
> 
@PolynomialDivision I was thinking changes like https://github.com/openwrt/openwrt/pull/12531/commits/c1f1e145e3a59639b23dc669ffa4ccd87ef00cad#diff-53028509d59d66b72d97f2686b4f03f8c79fd2d5cabd1701d75cd5aa9f9a4483 and https://github.com/openwrt/openwrt/pull/12531/commits/ceb7d5f200408a7c95dac8f162912833a25af143#diff-53028509d59d66b72d97f2686b4f03f8c79fd2d5cabd1701d75cd5aa9f9a4483 affect both 5.15 and 6.1 builds. If you're checking the upstream Kconfigs and Makefiles yourself to ensure things will build properly on both 5.15 and 6.1 then that's all fine. If not, then perhaps better to leave these updates to 6.1 merging/build testing, especially considering overlap with @Ansuel 's 6.1 branch (e.g. 188cbc370d415312f88e88b08bd9cf3f31b1482e).

File: target/linux/ramips/image/mt7621.mk [hunk 1]
```diff
@@ -9,10 +9,6 @@ DEFAULT_SOC := mt7621
 
 DEVICE_VARS += ELECOM_HWNAME LINKSYS_HWNAME DLINK_HWID
 
-ifdef CONFIG_LINUX_5_10
-  DTS_CPPFLAGS += -DDTS_LEGACY
```
Comment (DragonBluep) [inline target/linux/ramips/image/mt7621.mk:13]: This flag is used in "mt7621.dtsi", please clean it as well.

Comment (PolynomialDivision) [reply target/linux/ramips/image/mt7621.mk:13]: thanks!

File: package/kernel/qca-ssdk/patches/0001-SSDK-config-add-kernel-5.10.patch [hunk 2]
```diff
@@ -1,56 +0,0 @@
-From 6ee395e1cab32d1d4899d7846ea99e6e2dc0a9e4 Mon Sep 17 00:00:00 2001
```
Comment (1715173329) [inline package/kernel/qca-ssdk/patches/0001-SSDK-config-add-kernel-5.10.patch:1]: Need to refresh`0005-SSDK-config-add-kernel-5.15.patch` as it is built after this patch.

Comment (PolynomialDivision) [reply package/kernel/qca-ssdk/patches/0001-SSDK-config-add-kernel-5.10.patch:1]: Indeed. Thanks!

**Comment by PolynomialDivision** [comment]
> If you're checking the upstream Kconfigs and Makefiles yourself to ensure things will build properly on both 5.15 and 6.1 then that's all fine.

I just checked everything, if the symbols and names are still the same. However, I am not sure how to find the kmod dependencies of the different kmods. Is there a good way to check that?

**Comment by PolynomialDivision** [comment]
I would say this is ready.

**Comment by Ansuel** [comment]
@PolynomialDivision nice cleanup... I wonder if we should make a list of thing to check when these kind of thing has to happen. What do you think? And maybe put it in the wiki?

**Comment by PolynomialDivision** [comment]
> What do you think? And maybe put it in the wiki?

Yes of course. Documentation is always important. :)

**Comment by abajk** [comment]
It is still possible to remove some unused code:
~~~
$ grep -rni "LINUX_VERSION_CODE" target/ | grep -v ">= KERNEL_VERSION(5,15,0)"
...
target/linux/generic/files/drivers/net/phy/ar8216.c:894:#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 5, 0)
target/linux/generic/files/drivers/net/phy/ar8216.c:906:#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 5, 0)
target/linux/generic/files/drivers/net/phy/b53/b53_common.c:509:#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 5, 0)
target/linux/generic/files/drivers/net/phy/b53/b53_common.c:515:#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 5, 0)
target/linux/generic/files/block/partitions/fit.c:76:#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 11, 0)
target/linux/ramips/files/drivers/pinctrl/pinctrl-aw9523.c:389:#if LINUX_VERSION_CODE <= KERNEL_VERSION(5, 5, 0)
target/linux/ramips/files/drivers/pinctrl/pinctrl-aw9523.c:440:#if LINUX_VERSION_CODE <= KERNEL_VERSION(5, 5, 0)
target/linux/ramips/files/drivers/net/ethernet/ralink/mdio.c:67:#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0)
target/linux/ramips/files/drivers/net/ethernet/ralink/mdio.c:80:#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0)
target/linux/ramips/files/drivers/net/ethernet/ralink/soc_mt7620.c:147:#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0)
target/linux/ramips/files/drivers/net/ethernet/ralink/soc_mt7620.c:181:#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0)
target/linux/ramips/files/drivers/net/ethernet/ralink/mdio_rt2880.c:159:#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0)
target/linux/ramips/files/drivers/net/ethernet/ralink/mdio_rt2880.c:179:#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0)
target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_eth_soc.c:1099:#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0)

~~~

**Comment by neheb** [comment]
should probably add the ksmbd change here.

**Comment by PolynomialDivision** [comment]
> should probably add the ksmbd change here.

Which change?

**Comment by PolynomialDivision** [comment]
> It is still possible to remove some unused code:

@john-tho already suggsted that. :)

**Comment by neheb** [comment]
> > should probably add the ksmbd change here.
> 
> Which change?

https://github.com/openwrt/openwrt/pull/10296

Just keep the 5.15 changes.

**Comment by Ansuel** [comment]
@PolynomialDivision ok to merge this? any problem that may arise or it's safe?

**Comment by PolynomialDivision** [comment]
> ok to merge this?

I would say yes :)

> any problem that may arise or it's safe?

At least I don't see any issues.

**Pull request merged**